src/storage/store/block.rs

```diff
@@ -0,0 +1,134 @@
+use crate::proto::snapchain::Block;
+use crate::storage::db::{PageOptions, RocksDB, RocksDbTransactionBatch, RocksdbError};
+use prost::Message;
+use serde::Deserialize;
+use thiserror::Error;
+
+// TODO(aditi): This code definitely needs unit tests
+#[derive(Error, Debug)]
+pub enum BlockStorageError {
+    #[error(transparent)]
+    RocksdbError(#[from] RocksdbError),
+
+    #[error("Block missing header")]
+    BlockMissingHeader,
+
+    #[error("Block missing height")]
+    BlockMissingHeight,
+
+    #[error("Too many blocks in result")]
+    TooManyBlocksInResult,
+}
+
+pub const PAGE_SIZE_MAX: usize = 10_000;
+/** A page of messages returned from various APIs */
+pub struct BlockPage {
+    pub blocks: Vec<Block>,
+    pub next_page_token: Option<Vec<u8>>,
+}
+
+fn make_block_key(shard_index: u32, block_number: u64) -> Vec<u8> {
+    let mut key = [0u8; 12];
+    // Store the shard index as big-endian in the first 4 bytes
+    key[0..4].copy_from_slice(&shard_index.to_be_bytes());
+    // Store the block number in the remaining 8 bytes
+    key[4..12].copy_from_slice(&block_number.to_be_bytes());
+
+    key.to_vec()
```

Comment:
We should have a global prefix for the type of data we're storing, similar to the Hub's RootPrefix https://github.com/farcasterxyz/hub-monorepo/blob/7b3748906efa1a5b8f9b2755c2289d0fc8904ff8/apps/hubble/src/storage/db/types.ts#L23

Otherwise we're going to have prefix collisions when we store account state.

Reply:
Ah good point, added.